### DISCUSSION TITLE
Limitation to use variables in working_directory definition

### DISCUSSION THREAD

**Discussion by gingerbeard72** [batect/batect#1048] [Q&A] [answered]
I have been working on migrating our existing IaC 3-Musketeer based repos to batect. Part of the existing process sets the working_dir in compose to working_dir: /code/${SUBFOLDER}. My intention and hope was to use something similar with Batect but every combination of variable is not working for working_directory when defined in the environment or config_variables. 

The following is the container section of my batect.yml file;
    environment:
       subfolder: ${subfolder:-win_container}
    volumes:
      - local: <{batect.project_directory}
        container: /code
      - local: ~/.aws/
        container: /root/.aws/
    working_directory: /code/${SUBFOLDER}

On execution, this creates the folder $SUBFOLDER in the local project directory but not in the container which results in the container failing. I have also tried to use the cli options to define this variable and this is not working.

.\batect --config-var subfolder=win-container terraform_test
The config variable 'subfolder' set with --config-var has not been defined.

Are you able to confirm if this is a limitation of the working_directory definition or user error?

also, love batect and the simplicity.

**Comment by charleskorn** [comment, marked as answer]
Hi @gingerbeard72, great to hear that you're enjoying using Batect :)

At the moment, [expressions](https://batect.dev/docs/reference/config/expressions/) can only be used in a limited number of places, and `working_directory` is not one of them. #990 is an RFC to both replace the expression syntax with something more flexible and allow it to be used in more places, including `working_directory`.

As an interim workaround, you could configure an entrypoint script for your container that changes to the correct working directory based on the `subfolder` environment variable you're already defining, before then executing the command for your task.
